From 1880047f1bf0b045b253f8681d76b3cd6053ff22 Mon Sep 17 00:00:00 2001
From: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Date: Wed, 28 Feb 2024 17:44:56 +0100
Subject: [PATCH 0771/1695] drivers: stm32_bsec3: add
 stm32_bsec_self_hosted_debug_is_enabled()

The CoreSight devices can be accessed only when BSEC_DENR:DBGSWEN is
enabled.
With this API, the RCC driver will be able to return
TEE_ERROR_ACCESS_DENIED when attempting to enable debug clocks.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git fd5e77b2891fc87ead02db74049fd3855a33ff7e drivers: stm32_bsec3: add stm32_bsec_self_hosted_debug_is_enabled()]
Signed-off-by: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Change-Id: Idb00647ffc46f775901866081620819f58c84a59
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/363752
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/374139
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/drivers/stm32_bsec3.c        | 5 +++++
 core/include/drivers/stm32_bsec.h | 5 +++++
 2 files changed, 10 insertions(+)

diff --git a/core/drivers/stm32_bsec3.c b/core/drivers/stm32_bsec3.c
index b019c50df..d08e2691d 100644
--- a/core/drivers/stm32_bsec3.c
+++ b/core/drivers/stm32_bsec3.c
@@ -555,6 +555,11 @@ uint32_t stm32_bsec_read_debug_conf(void)
 	return io_read32(bsec_base() + BSEC_DENR) & BSEC_DENR_ALL_MASK;
 }
 
+bool stm32_bsec_self_hosted_debug_is_enabled(void)
+{
+	return stm32_bsec_read_debug_conf() & BSEC_DENR_DBGSWEN;
+}
+
 /*
  * bsec_get_version: return BSEC version.
  */
diff --git a/core/include/drivers/stm32_bsec.h b/core/include/drivers/stm32_bsec.h
index 8f3f2b2dd..b813b5979 100644
--- a/core/include/drivers/stm32_bsec.h
+++ b/core/include/drivers/stm32_bsec.h
@@ -184,6 +184,11 @@ bool stm32_bsec_can_access_otp(uint32_t otp_id);
  */
 bool stm32_bsec_nsec_can_access_otp(uint32_t otp_id);
 
+/*
+ * Return true if host-self debug is enabled.
+ */
+bool stm32_bsec_self_hosted_debug_is_enabled(void);
+
 /*
  * Find and get OTP location from its name.
  * @name: sub-node name to look up.
-- 
2.47.2

